### PR TITLE
Feature/wm 134 code context summary

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
This feature explains code context with GPT's API. 

Initially here on the frontend, we're only providing git + block of code info as parameters. The backend allows us to send ticketing and messaging system info as part of the prompt as well, which we'll do as we re-introduce those services. 

This PR sends the `codeContextSummary` to the sidebar's Provider, but doesn't actually render it yet. Should we wait until treeview is merged + we discuss that architecture to do so?

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  

## Acceptance
- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -212,12 +213,36 @@ export async function activate(context: vscode.ExtensionContext) {
           text: sortedPRs[0].title || parsedMessage,
         });
         */
-       
+
+        // get the start and end line of the selected block of code
+        let selectedStartLine = vscode.window.activeTextEditor?.selection.start;
```
Comment (EstebanDalelR) [inline src/extension.ts:218]: Is this the same as the params that this function receives?

File: src/extension.ts [hunk 2]
```diff
@@ -212,12 +213,36 @@ export async function activate(context: vscode.ExtensionContext) {
...
+        let selectedEndLine = vscode.window.activeTextEditor?.selection.end;
+
+        // get the text of the selected block of code
+        let selectedBlockOfCode = "";
+        let codeContextSummary = "";
```
Comment (EstebanDalelR) [inline src/extension.ts:223]: Why declare it so far from it's use?

File: src/extension.ts [hunk 3]
```diff
@@ -212,12 +213,36 @@ export async function activate(context: vscode.ExtensionContext) {
...
+          selectedBlockOfCode = selectedCode.replace(/(\r\n|\n|\r)/gm,"");
+        }
+          
+        codeContextSummary = await summarizeCodeConext({
+          pr_title: sortedPRs[0].title || parsedMessage,
+          pr_body: sortedPRs[0].body || parsedCommitObject.body,
+          block_of_code: selectedBlockOfCode || "",
+        });
+
         provider.sendMessage({
```
Comment (EstebanDalelR) [inline src/extension.ts:240]: Is provider defined?

File: src/utils/vscode/summarizeCodeContext.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+import { backendURL } from "../../constants";
+import analyticsReporter from "../vscode/reporter";
+
+export default async function summarizeCodeConext({
```
Comment (EstebanDalelR) [inline src/utils/vscode/summarizeCodeContext.ts:5]:
```suggestion
export default async function summarizeCodeContext({
```

File: src/utils/vscode/summarizeCodeContext.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    pr_body,
+    block_of_code,
+  }: {
+    pr_title: string;
+    pr_body: string;
+    block_of_code: string;
+  }) {
+    const codeContextSummary = await axios
+    .post(`${backendURL}/api/openai/summarizeCodeContext`, {
+    pr_title,
```
Comment (EstebanDalelR) [inline src/utils/vscode/summarizeCodeContext.ts:16]:
```suggestion
      pr_title,
```

File: src/utils/vscode/summarizeCodeContext.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      pr_body,
```
Comment (EstebanDalelR) [inline src/utils/vscode/summarizeCodeContext.ts:17]: So this endpoint does not validate the user at all?

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,10 @@ All notable changes to the "watermelon" extension will be documented in this fil
 
 Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.
 
+## [1.9.0]
```
Comment (EstebanDalelR) [inline CHANGELOG.md:7]: Is this version behind current?

**Review by EstebanDalelR — CHANGES_REQUESTED** [review]

File: src/utils/vscode/summarizeCodeContext.ts (hunk 6 above)
Comment (baristaGeek) [reply src/utils/vscode/summarizeCodeContext.ts:17]: What's the reason why it needs to?

File: src/utils/vscode/summarizeCodeContext.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
```
Comment (EstebanDalelR) [inline src/utils/vscode/summarizeCodeContext.ts:1]: Is this function used anywhere?

Comment (baristaGeek) [reply src/utils/vscode/summarizeCodeContext.ts:1]: Currently in the process of re-introducing it to the new UI

File: package.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "watermelon-tools",
   "description": "Watermelon, find all the historical context of your code",
-  "version": "1.7.2",
+  "version": "2.0.0",
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply package.json:4]: If you try committing code with "1.7.2" the handshake fails

Screen Shot 2023-01-25 at 12 46 56 PM

[the remark this answers is not in the document]
Comment (EstebanDalelR) [reply package.json:4]: IDK which handshake?

File: src/utils/vscode/summarizeCodeContext.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    block_of_code,
+    user_email
+  }: {
+    pr_title: string;
+    pr_body: string;
+    block_of_code: string;
+    user_email: string;
+  }) {
+    const codeContextSummary = await axios
+    // .post(`${backendURL}/api/openai/summarizeCodeContext`, {
```
Comment (EstebanDalelR) [inline src/utils/vscode/summarizeCodeContext.ts:17]: Weird URL

**Review by EstebanDalelR — APPROVED** [review]
LGTM.
Hoping users like it!

**Pull request merged**